subsys/nrf_security/src/drivers/cracen/cracen_sw/include/cracen_sw_aes_ccm.h

```diff
@@ -0,0 +1,231 @@
+					     psa_algorithm_t alg);
+
+/**
+ * @brief Set nonce for the software AES-CCM operation.
+ *
+ * @param[in,out] operation Pointer to the AEAD operation structure.
+ * @param[in] nonce Nonce value.
+ * @param[in] nonce_length Length of the nonce.
+ *
+ * @retval PSA_SUCCESS Nonce set successfully.
+ * @retval PSA_ERROR_INVALID_ARGUMENT Invalid nonce length (must be 7-13 bytes for CCM).
+ */
+psa_status_t cracen_sw_aes_ccm_set_nonce(cracen_aead_operation_t *operation, const uint8_t *nonce,
+					 size_t nonce_length);
+
+/**
+ * @brief Set lengths for the software AES-CCM operation.
+ *
+ * CCM requires the lengths of additional data and plaintext to be known upfront.
+ *
+ * @param[in,out] operation Pointer to the AEAD operation structure.
+ * @param[in] ad_length Length of additional data.
+ * @param[in] plaintext_length Length of plaintext.
+ */
+void cracen_sw_aes_ccm_set_lengths(cracen_aead_operation_t *operation, size_t ad_length,
+				   size_t plaintext_length);
+
+/**
+ * @brief Update the software AES-CCM operation with additional data.
+ *
+ * @param[in,out] operation Pointer to the AEAD operation structure.
+ * @param[in] input Additional data.
+ * @param[in] input_length Length of the additional data.
+ *
+ * @retval PSA_SUCCESS Additional data processed successfully.
+ * @retval PSA_ERROR_BAD_STATE AD already finalized or operation not initialized.
+ */
+psa_status_t cracen_sw_aes_ccm_update_ad(cracen_aead_operation_t *operation, const uint8_t *input,
+					 size_t input_length);
+
+/**
+ * @brief Update the software AES-CCM operation with new data.
+ *
+ * @param[in,out] operation Pointer to the AEAD operation structure.
+ * @param[in] input Input data (plaintext for encryption, ciphertext for decryption).
+ * @param[in] input_length Length of the input data.
+ * @param[out] output Output buffer (ciphertext for encryption, plaintext for decryption).
+ * @param[in] output_size Size of the output buffer.
+ * @param[out] output_length Pointer where to store the actual output length.
+ *
+ * @retval PSA_SUCCESS Data processed successfully.
+ * @retval PSA_ERROR_BUFFER_TOO_SMALL Output buffer too small.
+ */
+psa_status_t cracen_sw_aes_ccm_update(cracen_aead_operation_t *operation, const uint8_t *input,
+				      size_t input_length, uint8_t *output, size_t output_size,
+				      size_t *output_length);
+
+/**
+ * @brief Finish the software AES-CCM encryption operation.
+ *
+ * @param[in,out] operation Pointer to the AEAD operation structure.
+ * @param[out] ciphertext Output buffer for remaining ciphertext (typically none for CCM).
+ * @param[in] ciphertext_size Size of the ciphertext buffer.
+ * @param[out] ciphertext_length Pointer to store the actual ciphertext length.
+ * @param[out] tag Output buffer for the authentication tag.
+ * @param[in] tag_size Size of the tag buffer.
+ * @param[out] tag_length Pointer to store the actual tag length.
+ *
+ * @retval PSA_SUCCESS Operation finished successfully, tag generated.
```

Comment:
```suggestion
 * @retval PSA_SUCCESS Operation finished successfully; tag generated.
```
Or `:`, as mentioned earlier.